e2e/test/scenarios/dashboard-filters/dashboard-chained-filters.cy.spec.js

```diff
@@ -141,71 +144,106 @@ describe("scenarios > dashboard > chained filter", () => {
     });
   }
 
-  it.skip("should work for all field types (metabase#15170)", () => {
-    // Change Field Types for the following fields
-    cy.request("PUT", `/api/field/${PRODUCTS.ID}`, {
-      special_type: null,
-    });
-
-    cy.request("PUT", `/api/field/${PRODUCTS.EAN}`, {
-      special_type: "type/PK",
-    });
+  it(
+    "should work for all field types (metabase#15170)",
+    { tags: "@external" },
+
+    () => {
+      const dialect = "postgres";
+      const TEST_TABLE = "many_data_types";
+
+      resetTestTable({ type: dialect, table: TEST_TABLE });
+      restore(`${dialect}-writable`);
+      cy.signInAsAdmin();
+      resyncDatabase({ tableName: TEST_TABLE, tableAlias: "testTable" });
+
+      cy.get("@testTable").then(testTable => {
+        const testTableId = testTable.id;
+        const uuidFieldId = testTable.fields.find(
+          field => field.name === "uuid",
+        ).id;
+        const idFieldId = testTable.fields.find(
+          field => field.name === "id",
+        ).id;
+
+        cy.wrap(testTableId).as("testTableId");
+        cy.wrap(uuidFieldId).as("uuidFieldId");
+
+        cy.log(
+          "Mimics that UUID is the table's primary key, so we could map dashboard ID parameter to UUID",
+        );
+        cy.request("PUT", `/api/field/${idFieldId}`, {
+          semantic_type: null,
+        });
 
-    cy.createQuestion({
-      name: "15170",
-      query: { "source-table": PRODUCTS_ID },
-    }).then(({ body: { id: QUESTION_ID } }) => {
-      cy.createDashboard().then(({ body: { id: DASHBOARD_ID } }) => {
-        // Add filter to the dashboard
-        cy.request("PUT", `/api/dashboard/${DASHBOARD_ID}`, {
-          parameters: [
-            {
-              id: "50c9eac6",
-              name: "ID",
-              slug: "id",
-              type: "id",
-            },
-          ],
+        cy.request("PUT", `/api/field/${uuidFieldId}`, {
+          semantic_type: "type/PK",
         });
+      });
+
+      cy.then(function () {
+        const TEST_TABLE_ID = this.testTableId;
+        const UUID_FIELD_ID = this.uuidFieldId;
+
+        cy.createQuestion({
```

Comment:
From this point, the test body is almost the same, but I change the question source and field to map with the dashboard ID parameter.